components/clarinet-format/src/formatter/mod.rs

```diff
@@ -123,10 +132,9 @@ impl<'a> Aggregator<'a> {
         // this handles if we're formatting a section of code rather than the whole file
         let indentation_level = self
             .source
-            .chars()
-            .take_while(|c| c.is_whitespace())
-            .count();
-        self.format_source_exprs(self.pse, &self.source[..indentation_level])
+            .map_or(0, |s| s.chars().take_while(|c| c.is_whitespace()).count());
```

Comment:
This might be wrong. I think we'd have to check the PSE `span().start_column` to get an accurate indentation level

Reply:
`self.pse[0].span().start_column - 1` I think. It's possible this is just easier than counting whitespaces from source anyway. I believe there are tests for this, so it'd be good for the formatter. A question might be how LSP passes the PSEs in though. Worth trying in the browser to make sure